### PR TITLE
Convert import delimited file to Top Component 

### PR DESCRIPTION
**Pull request by aldebaran30701**
### Description of the Change
This PR fixes the problem relating to #613. It now listens to the graph for changes and updates the combo box.
A visual refactor of the view to allow easier usage. 
* Drop down for attribute adding
* Scroll-able pane for attribute adding
* New dark mode
* Invert colors depending on attribute color background

Fix a bug with the add all attributes not refreshing the attribute pane

### Alternate Designs
Adding class dependencies to allow a listening framework to be added. - not followed as it was introducing new concepts to the importer.

### Why Should This Be In Core?
Visual Improvement and bug fixes

### Benefits
Visual Improvement and bug fixes

Before:
import old

After:
image

### Possible Drawbacks
none realised

### Verification Process
Import testing and view loading

### Applicable Issues
#613 
Found #805 through testing.

### CODE REVIEW

**Comment by aldebaran30701** [comment]
Will be fixing some of the Sonar bugs at a later point today. 👍

**Comment by arcturus2** [comment]
@aldebaran30701 you have delivered the goods once again mate. This is really cool and to me makes perfect sense to be a top component. I am yet to test it out but the code etc looks good. It is nice to see the delimited file importer getting some good love.

**Comment by aldebaran30701** [comment]
As per feedback - 
* Abort button has been removed as it served no purpose on the view.
* Drop down combo box now holds the previous selection when a new graph is made.

I checked the feasibility of switching to a Go/Stop button similar to the Data Access View - but it will need a bit of work to get it working. This is because it runs in a plugin and it's behaviour might be confusing when running multiple run tabs of imports. Currently the import button is shared with all run tabs.

File: CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/SourcePane.java [hunk 1]
```diff
@@ -302,4 +276,45 @@ void update(final ImportController importController) {
...
         return graphComboBox.getSelectionModel().getSelectedItem();
     }
+
+    /**
+     * Helper method to update the destination graph combo box. Also used for
+     * value loading when this pane is being initialized.
+     */
+    public void updateDestinationGraphCombo() {
+        // previousDestinationObject is the previously selected item in the combobox
+        ImportDestination<?> previousDestinationObject = graphComboBox.getSelectionModel().getSelectedItem();
```
Comment (antares1470) [inline CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/SourcePane.java:286]: final

**Review by antares1470 — CHANGES_REQUESTED** [review]

**Review by antares1470 — APPROVED** [review]

**Review by arcturus2 — APPROVED** [review]

File: CoreUtilities/src/au/gov/asd/tac/constellation/utilities/color/ConstellationColor.java [hunk 2]
```diff
@@ -330,6 +330,65 @@ public static ConstellationColor fromHtmlColor(final String color) {
...
     }
 
+    /**
+     * Get a contrasting color based on YIQ values. Will return BLACK if the
+     * passed in color is null.
+     *
+     * @param color HTML color to contrast with
+     * @return the ConstellationColor which contrasts with color.
+     */
+    public static ConstellationColor getContrastHtmlColor(final String color) {
```
Comment (arcturus2) [inline CoreUtilities/src/au/gov/asd/tac/constellation/utilities/color/ConstellationColor.java:340]: These methods are great! Can you please document these additional methods in `CHANGELOG.md`?

File: CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/resources/delimited-importer.css [hunk 3]
```diff
@@ -0,0 +1 @@
+/* Delimited Importer */
```
Comment (arcturus2) [inline CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/resources/delimited-importer.css:1]: This file is empty so can it be removed?

File: CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/SourcePane.java [hunk 4]
```diff
@@ -140,28 +141,29 @@ public SourcePane(final ImportController importController) {
...
+            final List<File> newFiles = fileChooser.showOpenMultipleDialog(SourcePane.this.getScene().getWindow());
 
             if (newFiles != null) {
                 if (!newFiles.isEmpty()) {
                     DEFAULT_DIRECTORY = newFiles.get(0).getParentFile();
                     SourcePane.this.importFileParserComboBox.setDisable(true);
                 }
                 final ObservableList<File> files = FXCollections.observableArrayList(fileListView.getItems());
-                String warningMsg = "The following files could not be parsed and have been excluded from import set:";
+                final StringBuilder sb = new StringBuilder();
```
Comment (arcturus2) [inline CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/SourcePane.java:165]: :+1:

**Review by arcturus2 — CHANGES_REQUESTED** [review]
I've made a couple of minor comments for you @aldebaran30701

File: CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/resources/delimited-importer.css (hunk 3 above)
Comment (aldebaran30701) [reply CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/resources/delimited-importer.css:1]: Yep I think so.

File: CoreUtilities/src/au/gov/asd/tac/constellation/utilities/color/ConstellationColor.java (hunk 2 above)
Comment (aldebaran30701) [reply CoreUtilities/src/au/gov/asd/tac/constellation/utilities/color/ConstellationColor.java:340]: I already had I believe. Do you want me to further expand on this?
`Added utility methods to `ConstellationColor` which assist with getting inverse colors.`

Comment (arcturus2) [reply CoreUtilities/src/au/gov/asd/tac/constellation/utilities/color/ConstellationColor.java:340]: > 
> 
> I already had I believe. Do you want me to further expand on this?
> `Added utility methods to `ConstellationColor` which assist with getting inverse colors.`

No that's perfect, I missed it.

**Review by arcturus2 — APPROVED** [review]

**Pull request merged**